sklearn/model_selection/_search.py

```diff
@@ -618,15 +618,16 @@ def fit(self, X, y=None, groups=None, **fit_params):
             if self.refit is not False and (
                     not isinstance(self.refit, six.string_types) or
                     # This will work for both dict / list (tuple)
-                    self.refit not in scorers):
-                raise ValueError("For multi-metric scoring, the parameter "
-                                 "refit must be set to a scorer key "
-                                 "to refit an estimator with the best "
-                                 "parameter setting on the whole data and "
-                                 "make the best_* attributes "
-                                 "available for that metric. If this is not "
-                                 "needed, refit should be set to False "
-                                 "explicitly. %r was passed." % self.refit)
+                    self.refit not in scorers) and not callable(self.refit):
+                    raise ValueError("For multi-metric scoring, the parameter "
```

Comment:
this appears to have the wrong indentation. It cannot have the same indent as the previous line with the `:`